### PR TITLE
Add support for EKS platform

### PR DESCRIPTION
**Pull request by johnctitus**
Updates the ebs_device_map local variable to include Amazon ECS and EKS

Adds local variable for EKS userdata snippet and adds reference in amazon userdata template

Add variables for EKS cluster name and additional bootstrapping parameters.  Corrects default value of `ec2_scale_down_adjustment` variable.

Adds header to top of main.tf to generate README usage example and summary.

Adds IAM role name to outputs, for use by EKS module.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+# aws-terraform-ec2_asg
+
+This module creates one or more autoscaling groups.
+
+## Basic Usage
+
+```
+module "rds" {
```
Comment (jonathan-madison) [inline README.md]: Should this be labeled RDS?

File: main.tf [hunk 2]
```diff
@@ -1,3 +1,25 @@
+/**
+ * # aws-terraform-ec2_asg
+ *
+ *This module creates one or more autoscaling groups.
+ *
+ *## Basic Usage
+ *
+ *```
+ *module "rds" {
```
Comment (jonathan-madison) [inline main.tf]: Another instance of "rds"

File: README.md (hunk 1 above)
Comment (johnctitus) [reply README.md]: Corrected

File: main.tf (hunk 2 above)
Comment (johnctitus) [reply main.tf]: Corrected

**Pull request merged**